### ISSUE TITLE
Index into NamedTuple with ::Colon

### ISSUE THREAD

**Issue by kescobo** [JuliaLang/julia#47821]
I should probably have put this in in #38878.

```
julia> nt = (a = 1, b = 2, c = 3);

julia> nt[[:a, :b]]
(a = 1, b = 2)

julia> nt[:]
ERROR: MethodError: no method matching getindex(::NamedTuple{(:a, :b, :c), Tuple{Int64, Int64, Int64}}, ::Colon)
```

**Comment by Moelf** [comment]
what's the expected behavior?
```julia
getindex(a::NamedTuple, ::Colon) = a
```

?

**Comment by kescobo** [comment]
Yeah, I think so.

**Comment by Moelf** [comment]
add `good first issue`?

**Comment by kescobo** [comment]
@x0samnan I would take a look at [this blog post](https://kshyatt.github.io/post/firstjuliapr/). It's a bit dated, but is a good starting point.

Though I was planning to tackle this since it's my fault it got left out :shrug:, there are [lot of good first issues ](https://github.com/JuliaLang/julia/labels/good%20first%20issue) you could check out :-)

**Comment by kescobo** [comment]
Closed in #47842
